thunder/core/prims.py

```diff
@@ -1814,10 +1814,13 @@ def _put_grad_meta(grad_for: Number | NumberProxy | TensorProxy, grad: Number |
     return None
 
 
+# PUT_GRAD is a sink node with side effects that updates Tensor.grad. It needs
+# DONT_DCE tag to avoid removal in DCE pass.
```

Comment:
Thanks for the explanation! This could be the reason `use_dce=False` was set in vjp_utils.py file